tests/scalers/aws/aws_cloudwatch_ignore_null_values_false/aws_cloudwatch_ignore_null_values_false_test.go

```diff
@@ -0,0 +1,197 @@
+        metricName: {{.CloudWatchMetricName}}
+        targetMetricValue: "1"
+        minMetricValue: "1"
+        ignoreNullValues: "false"
+        metricCollectionTime: "120"
+        metricStatPeriod: "60"
+`
+)
+
+var (
+	testNamespace                  = fmt.Sprintf("%s-ns", testName)
+	deploymentName                 = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName               = fmt.Sprintf("%s-so", testName)
+	secretName                     = fmt.Sprintf("%s-secret", testName)
+	cloudwatchMetricName           = fmt.Sprintf("cw-%d", GetRandomNumber())
+	awsAccessKeyID                 = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey             = os.Getenv("TF_AWS_SECRET_KEY")
+	awsRegion                      = os.Getenv("TF_AWS_REGION")
+	cloudwatchMetricNamespace      = "DoesNotExist"
+	cloudwatchMetricDimensionName  = "dimensionName"
+	cloudwatchMetricDimensionValue = "dimensionValue"
+	minReplicaCount                = 0
+)
+
+// This test is to verify that the scaler results in an error state when
+// the metric query returns null values and the ignoreNullValues is set to false.
+func TestCloudWatchScalerWithIgnoreNullValuesFalse(t *testing.T) {
+	ctx := context.Background()
+
+	// setup cloudwatch
+	cloudwatchClient, err := cloudwatch.NewClient(ctx, awsRegion, awsAccessKeyID, awsSecretAccessKey, "")
+	require.Nil(t, err, "error creating cloudwatch client")
+
+	// check that the metric in question is not already present, and is returning
+	// an empty set of values.
+	metricQuery := cloudwatch.CreateMetricDataInputForEmptyMetricValues(cloudwatchMetricNamespace, cloudwatchMetricName, cloudwatchMetricDimensionName, cloudwatchMetricDimensionValue)
+	metricData, err := cloudwatch.GetMetricData(ctx, cloudwatchClient, metricQuery)
+	require.Nil(t, err, "error getting metric data")
+	require.Nil(t, cloudwatch.ExpectEmptyMetricDataResults(metricData), "metric data should be empty")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	kedaClient := GetKedaKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+	defer DeleteKubernetesResources(t, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
+
+	// check that the scaledobject is in paused state
+	FailIfScaledObjectStatusNotReachedWithTimeout(t, kedaClient, testNamespace, scaledObjectName, 2*time.Minute, v1alpha1Api.ConditionPaused)
+
+	// check that the deployment did not scale, as the metric query is returning
+	// null values and the scaledobject is receiving errors, the deployment
+	// should not scale.
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
```

Comment:
In order to assert that replica count doesn't change, we have to use this other func:
```suggestion
	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minReplicaCount, 60)
```
It will test the workload every second during the given time (60 seconds) to check that the current value is the given one